Update NuGet Audit pruning feature description (#48317)
dotnet/docs: docs/core/whats-new/dotnet-10/sdk.md

## Pruning of framework-provided package references

> Starting in .NET 10, the [NuGet Audit](/nuget/concepts/auditing-packages) feature can now [prune framework-provided package references](/nuget/consume-packages/package-references-in-project-files#prunepackagereference) that aren't used by the project. 

Reviewer: ```suggestion
Starting in .NET 10, the [NuGet Audit](/nuget/concepts/auditing-packages) feature can [prune framework-provided package references](/nuget/consume-packages/package-references-in-project-files#prunepackagereference) that aren't used by the project.
```

This feature is enabled by default for all frameworks of a project that targets >= .NET 10.0 in the latest SDK.
This change helps reduce the number of packages that are restored and analyzed during the build process, which can lead to faster build times and reduced disk space usage. It also can lead to a reduction in false positives from NuGet Audit and other dependency-scanning mechanisms.

When this feature is enabled, you might see a reduction in the contents of your applications' generated *.deps.json* files. Any package references supplied by the .NET runtime are automatically removed from the generated dependency file.

> When a direct package reference is within the pruning range, [`PrivateAssets="all"` and `IncludeAssets="none"` will be applied](/nuget/consume-packages/package-references-in-project-files#how-prunepackagereference-works).

Reviewer: ```suggestion
When a direct package reference is within the pruning range, [`PrivateAssets="all"` and `IncludeAssets="none"` are applied](/nuget/consume-packages/package-references-in-project-files#how-prunepackagereference-works).
```
